code/modules/antagonists/wraith/objs/machinery/harbinger_guard.dm

```diff
@@ -0,0 +1,86 @@
+/obj/machinery/wraith/harbinger_guard
+	name = "Ghostly sentry"
+	icon = 'icons/obj/wraith_objects.dmi'
+	icon_state = "harbinger_guard"
+	desc = "An immobile guardian, ready to slam on anyone getting close to it."
+	anchored = 1
+	density = 1
+	_health = 25
+	var/mob/living/intangible/wraith/wraith_harbinger/master = null
+	var/datum/light/light
+	var/cooldown_attack = 8 SECONDS
+	var/attack_delay = 1.2 SECONDS
+
+	New(var/turf/T, var/mob/living/intangible/wraith/wraith_harbinger/W)
+		src.set_loc(T)
+		src.visible_message("<span class='alert'>A [src] manifests!</b></span>")
+		light = new /datum/light/point
+		light.set_brightness(0.1)
+		light.set_color(150, 40, 40)
+		light.attach(src)
+		light.enable()
+		START_TRACKING
+		src.master = W
+		setup_use_proximity()
+		src.alpha = 0
+		animate(src, alpha=255, time=2 SECONDS)
+		ON_COOLDOWN(src, "attack", 3 SECONDS)
+		..()
+
+	process()
+		if (GET_COOLDOWN(src, "attack"))
+			return
+
+		for (var/mob/living/M in range(src, 2))
+			if ((ishuman(M) || issilicon(M)) && !isdead(M))
+				ON_COOLDOWN(src, "attack", cooldown_attack)
+				src.attack(M)
+				UpdateIcon()
+				break
+
+
+	disposing()
+		if (src.master != null)
+			src.master.guard_amount --
+		. = ..()
+
+	attackby(obj/item/P as obj, mob/living/user as mob)
+		src._health -= P.force
+		attack_particle(user,src)
```

Comment:
Isn't there some proc for modifying _health? If not, there probably should be.

Reply:
It seems there isnt, and that each machinery has their own way of treating damage. Would you prefer that i bundled a proc for machineries to treat damage in this pr, or should it be atomized?

Reply:
Actually, changeHealth() does exactly that, and also triggers destruction when health reaches zero. Implemented it and added it to the portal too.